### ISSUE TITLE
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: rpc error on method: finality_update

### ISSUE THREAD

**Issue by ltfschoen** [a16z/helios#272]
i was running a node with 
```
cargo run -- \
  --network goerli \
  --consensus-rpc http://testing.prater.beacon-api.nimbus.team \
  --execution-rpc https://ethereum-goerli-rpc.allthatnode.com \
  --checkpoint 0x3754cde3eabb1815c15d78ef27a279d4abbd3ece0f720535acd5ca80b3a16f91
```

and then got this error:
```
[2023-09-02T10:58:02Z INFO  consensus::consensus] updated head               slot=6428689  confidence=78.12%  age=00:00:00:14
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: rpc error on method: finality_update, message: error sending request for url (http://testing.prater.beacon-api.nimbus.team/eth/v1/beacon/light_client/finality_update): connection closed before message completed

Location:
    consensus/src/rpc/nimbus_rpc.rs:66:19', /eip-x/helios/consensus/src/consensus.rs:105:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

i simply started the node again and it worked again...

i'm guessing it may have been caused because i was using a public endpoint rather than Alchemy? or is it because i need to configure the light client to connection with the provider to have a different polling interval / retry timeout / network check timeout, or similar?

**Comment by ncitron** [comment]
The request may have failed for many reasons on the rpc end. This does bring up a good point though that we should be using a retry policy for all api requests helios makes so that a simple connection failure will not cause Helios to crash.

I can work on a fix this week.

**Comment by muang0** [comment]
@ncitron Would you like me to look at this as a good first issue? I'm interested in contributing to this repo.

**Comment by ncitron** [comment]
That would be great @muang0!

**Comment by muang0** [comment]
@ncitron any concerns with using the [backoff](https://crates.io/crates/backoff) crate? Took an initial [pass](https://github.com/a16z/helios/compare/master...muang0:helios:master#diff-56f8bb55d7972a5310efa5cc9b974ca2b3c4ac8b38ceca8ef04a64b6b3c8fbf1) at it.

**Comment by ncitron** [comment]
That crate looks fine.

**Comment by ncitron** [comment]
Closed by #274
